src/renderer.zig

```diff
@@ -887,8 +887,11 @@ pub const MeshSelection = struct { // MARK: MeshSelection
 	pub var selectedBlockPos: ?Vec3i = null;
 	var lastSelectedBlockPos: Vec3i = undefined;
 	var currentBlockProgress: f32 = 0;
-	var currentSwingProgress: f32 = 0;
-	var currentSwingTime: f32 = 0;
+	pub var currentSwingProgress: f32 = 0;
+	pub var hasHit: bool = false;
+	pub const swingProgressHitTime: f32 = 0.4;
+	pub var currentSwingTime: f32 = 0;
+	pub var undertimeFactor: f32 = 1.0;
```

Comment:
Not sure what you did, but under-time doesn't appear to work as intendended anymore.
Undertime is supposed to be applied every time a block is broken, but for you it only seems to happen every other block for tools with high damage:

https://github.com/user-attachments/assets/ef5c27a1-8b79-49a6-83a3-ef3bf182ab69